ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/T8nExecutor.java

```diff
@@ -389,13 +388,28 @@ static T8nResult runTest(
         tracer = tracerManager.getManagedTracer(transactionIndex, transaction.getHash());
         tracer.tracePrepareTransaction(worldStateUpdater, transaction);
         tracer.traceStartTransaction(worldStateUpdater, transaction);
+        BlockHashLookup blockHashLookup =
+            protocolSpec.getBlockHashProcessor().createBlockHashLookup(blockchain, blockHeader);
+        if (blockHashLookup instanceof BlockchainBasedBlockHashLookup) {
+          // basically t8n test cases for blockhash are broken and one cannot create a blockchain
+          // from them so need to
+          // add in a manual BlockHashLookup
+          blockHashLookup =
+              (__, blockNumber) -> {
+                if (referenceTestEnv.getNumber() - blockNumber > 256L
+                    || blockNumber >= referenceTestEnv.getNumber()) {
+                  return Hash.ZERO;
+                }
+                return referenceTestEnv.getBlockhashByNumber(blockNumber).orElse(Hash.ZERO);
+              };
```

Comment:
This happens because before the block range validation was done inside of the BlockHashOperation. This check was moved to the BlockHashLookup implementation so that is why we have to have the same block range check here. Is my understanding correct?

Reply:
yup, the same happens with https://github.com/hyperledger/besu/pull/7971/files/e310147d8efd8bf5115f231b050c2f6b20f1512a#r1878989411 but my comment is still correct. From what I remember the test is broken because the fixtures for the blockchain state are not valid. We should be able to build the state as these are integration tests and we have a fully working `BlockHashProcessor`. But I don't think there's a way to change them as they were generated with some tool and changing them would imply recalculating hashes, so we have to add a manual BlockHashLookup.